### PR TITLE
add support for reports

### PR DESCRIPTION
**Pull request by dcavanagh**

### CODE REVIEW

File: src/Client.ts [hunk 1]
```diff
@@ -52,4 +56,41 @@ export class Client {
       token: accessToken,
     });
   }
+
+  public async getReports(query?: GetStationsQuery): Promise<Array<Report>> {
```
Comment (andydevlinsmith) [inline src/Client.ts]: Should this be GetReportsQuery? ^^

**Review by andydevlinsmith — APPROVED** [review]

File: src/Client.ts (hunk 1 above)
Comment (dcavanagh) [reply src/Client.ts]: We named the above query methods `getData` and `getStations`. It returns an `Array<Report>` so I think the name works. We could change them to `queryData`, `queryStations`, & `queryData`?

Comment (ianmetcalf) [reply src/Client.ts]: @dcavanagh I think he means the type of the first argument.

File: README.md [hunk 2]
```diff
@@ -90,9 +90,9 @@ Stations are returned in alphabetical order by station name.
         "filename": "Test_OneMin.dat",
         "is_stale": false,
         "headers": {
-          "columns": "Ambient_Temp",
-          "units": "Deg_C",
-          "processing": "Avg"
+          "columns": ["Ambient_Temp"],
+          "units": ["Deg_C"],
+          "processing": ["Avg]"
```
Comment (ianmetcalf) [inline README.md:96]:
```suggestion
          "processing": ["Avg"]
```

File: src/Client.ts [hunk 3]
```diff
@@ -1,4 +1,6 @@
-import {STATIONS_URL, DATA_URL} from './config';
+import {
+  STATIONS_URL, DATA_URL, REPORTS_URL, EXPORTS_URL,
```
Comment (ianmetcalf) [inline src/Client.ts:2]:
```suggestion
  STATIONS_URL,
  DATA_URL,
  REPORTS_URL,
  EXPORTS_URL,
```

File: README.md [hunk 4]
```diff
@@ -209,3 +209,134 @@ Takes a post data payload object as an argument and uploads it to the cloud.
...
+]
+```
+
+### Get Report
+
+```typescript
+client.getReport(reportKey: string): Promise<Report>
+```
+
+Takes a report key as an argument and returns the report with a url to the report artifact.
```
Comment (ianmetcalf) [inline README.md:288]: We should probably include a note that the URL will only be valid for an hour.

File: README.md [hunk 5]
```diff
@@ -209,3 +209,134 @@ Takes a post data payload object as an argument and uploads it to the cloud.
...
+}
+```
+
+### Get Data Export
+
+```typescript
+client.getDataExport(dataExportKey: string): Promise<DataExport>
+```
+
+Takes an export key as an argument and returns the export with a url to the export artifact.
```
Comment (ianmetcalf) [inline README.md:324]: We should probably include a note that the URL will only be valid for an hour.

**Review by ianmetcalf — COMMENTED** [review]
Couple items that need fixing

File: src/Client.ts (hunk 1 above)
Comment (dcavanagh) [reply src/Client.ts]: :disappointed: Fixed

**Comment by ianmetcalf** [comment]
closed in favor of #28

**Pull request closed without merging**